### PR TITLE
fix(meta): Fix sink_into_table causes mv's depended_subscription_ids to be empty.

### PR DESCRIPTION
**Pull request by xxhZs**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

1. Fixed the get_mv_depended_subscriptions method not really querying all subscription
2. Use old_table_fragments.table_id to get subscriptions instead of table_fragments.table_id when replacing table

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/meta/src/controller/catalog.rs [hunk 1]
```diff
@@ -2670,6 +2670,29 @@ impl CatalogController {
...
+    ) -> MetaResult<HashMap<risingwave_common::catalog::TableId, HashMap<u32, u64>>> {
+        let inner = self.inner.read().await;
+        let subscription_objs = Subscription::find()
+            .find_also_related(Object)
+            .all(&inner.db)
+            .await?;
+        let mut map = HashMap::new();
+        for subscription in subscription_objs
+            .into_iter()
+            .map(|(subscription, obj)| ObjectModel(subscription, obj.unwrap()).into())
```
Comment (wenym1) [inline src/meta/src/controller/catalog.rs:2684]: Shall we add comment here why it's safe to `unwrap`?

File: e2e_test/subscription/main.py [hunk 2]
```diff
@@ -304,6 +304,27 @@ def test_cursor_fetch_n():
...
+    create_table_subscription()
+    conn = psycopg2.connect(
+        host="localhost",
+        port="4566",
+        user="root",
+        database="dev"
+    )
+
+    execute_insert("declare cur subscription cursor for sub2",conn)
+    execute_insert("insert into t2 values(1,1)",conn)
```
Comment (wenym1) [inline e2e_test/subscription/main.py:318]: We may manually trigger ad-hoc recovery, or run a sink-into-table, to cover the error case.

File: src/meta/src/stream/stream_manager.rs [hunk 3]
```diff
@@ -504,8 +511,13 @@ impl GlobalStreamManager {
             existing_locations,
         }: ReplaceTableContext,
     ) -> MetaResult<()> {
-        self.build_actors(&table_fragments, &building_locations, &existing_locations)
-            .await?;
+        self.build_actors(
```
Comment (wenym1) [inline src/meta/src/stream/stream_manager.rs:514]: IIUC, for `replace_table`, the table_id of the new TableFragment should be the same as the `old_table_fragments`. You can add an `assert_eq` to verify this. If this is true, in `build_actors`, we can always use the `table_id` of `table_fragments` and we don't need to pass the separate table_id to the method.

File: e2e_test/subscription/main.py (hunk 2 above)
Comment (xxhZs) [reply e2e_test/subscription/main.py:318]: add sink into table in create_table_and_subscription.slt

**Review by wenym1 — APPROVED** [review]
LGTM

**Pull request merged**